src/inc/tetris_component.h

```diff
@@ -0,0 +1,14 @@
+#ifndef TETRIS_COMPONENT_H
+#define TETRIS_COMPONENT_H
+
+#include "game_component.h"
+
+class TetrisComponent : public GameComponent
+{
+public:
+  
+    virtual void accept( ComponentVisitor& visitor ) = 0;
```

Comment:
The ComponentVisitor interface contains logic specific to Tetris; I have been investigating ways to remove those details from the engine for the past few days. Runtime polymorphism is required to use the visitor pattern implemented in the game, so I could not make ComponentVisitor a template. However, this solution allows for a complete separation of concerns.